src/utils/snaps.ts

```diff
@@ -0,0 +1,102 @@
+import type { ExternalProvider } from '@ethersproject/providers'
+import detectEthereumProvider from '@metamask/detect-provider'
+import assert from 'assert'
+import { getConfig } from 'config'
+// Snaps utils re-imported from http://github.com/shapeshift/metamask-snaps-adapter until it's actually ready
+// We should eventually interface with hdwallet and can remove all this fluff
+
+export interface WalletEnableResult {
+  accounts: string[]
+  permissions: any[]
+  snaps: any
+  errors?: Error[]
+}
+interface EnableShapeShiftSnapResult {
+  success: boolean
+  message: WalletEnableResult
+}
+
+type Provider = Omit<ExternalProvider, 'request'> & {
+  request?: (request: { method: string; params?: any }) => Promise<any>
+}
+
+export const shapeShiftSnapInstalled = async (snapId: string): Promise<boolean> => {
```

Comment:
This is the root cause of failing send tests. This was copied and pasted from the snaps repo, but the control flow is totally derp here. We should not throw, we should try, catch, log errors, and return false.

Reply:
4fb206ef600ce0dbe5b3c4fe15d5bb4a7faf2372 and ee0f9be05829d43524087a1352f3006a41ea57b9